### ISSUE TITLE
Error when calling pgo backup (pgbackrest) multiple times [V 4.0.1]

### ISSUE THREAD

**Issue by ajoyej** [CrunchyData/postgres-operator#964]
Hello,
If we run multiple times a pgo backup command (pgbackrest) on a given cluster, only the 1st backup will be executed with success. Any other attempts will fail.

Looking at the log, this is due to the fact that the backup jobs already exists and cannot be recreated. 
--> if we manually delete the job (eg : backrest-backup-cbcd70 job), it is possible to execute another backup... but the next one will fail again.

Note:
If we run backups using the  pgo scheduler this works without any issues.
```
pgo create schedule cbcd60 --schedule="*/5 * * * *" --schedule-type=pgbackrest --pgbackrest-backup-type=full -n cfae-postgres-testing
```

------
Step to reproduce
--------------------
Initial part:
```
# Creating the cluster
$ pgo create cluster cbcd70 -n cfae-postgres-testing
created Pgcluster cbcd70
workflow id df3269fb-12d5-46cd-a079-0daa94d8a6fc

# testing the cluster
$ pgo test cbcd70 -n cfae-postgres-testing

cluster : cbcd70
        psql -p 5432 -h 172.24.199.244 -U postgres postgres is Working
        psql -p 5432 -h 172.24.199.244 -U postgres userdb is Working
        psql -p 5432 -h 172.24.199.244 -U primaryuser postgres is Working
        psql -p 5432 -h 172.24.199.244 -U primaryuser userdb is Working
        psql -p 5432 -h 172.24.199.244 -U testuser postgres is Working
        psql -p 5432 -h 172.24.199.244 -U testuser userdb is Working

# show the status of backup
$ pgo show backup cbcd70 -n cfae-postgres-testing

backrest : cbcd70

Storage Type: local
stanza: db
    status: error (no valid backups)
    cipher: none

    db (current)
        wal archive min/max (11-1): none present
```

first backup [OK]:
```
# backup 
$ pgo backup cbcd70 --backup-opts="--type=full" -n cfae-postgres-testing
created Pgtask backrest-backup-cbcd70

# job and pod
$ kubectl get jobs backrest-backup-cbcd70
NAME                     DESIRED   SUCCESSFUL   AGE
backrest-backup-cbcd70   1         1            1m

$ kubectl get pod backrest-backup-cbcd70-pjjw2
NAME                           READY     STATUS      RESTARTS   AGE
backrest-backup-cbcd70-pjjw2   0/1       Completed   0          2m

# logs
time="2019-08-28T12:53:36Z" level=debug msg="created Job backrest-backup-cbcd70"
time="2019-08-28T12:53:36Z" level=debug msg="[PgtaskController] onUpdate ns=cfae-postgres-testing /apis/crunchydata.com/v1/namespaces/cfae-postgres-testing/pgtasks/backrest-backup-cbcd70"
time="2019-08-28T12:53:36Z" level=debug msg="JobController: onAdd ns=cfae-postgres-testing jobName=/apis/batch/v1/namespaces/cfae-postgres-testing/jobs/backrest-backup-cbcd70"
time="2019-08-28T12:53:36Z" level=debug msg="[PodController] OnAdd ns=cfae-postgres-testing /api/v1/namespaces/cfae-postgres-testing/pods/backrest-backup-cbcd70-pjjw2"
time="2019-08-28T12:53:36Z" level=debug msg="job pod found [backrest-backup-cbcd70-pjjw2]"
time="2019-08-28T12:53:36Z" level=debug msg="[PodController] onUpdate ns=cfae-postgres-testing /api/v1/namespaces/cfae-postgres-testing/pods/backrest-backup-cbcd70-pjjw2"
time="2019-08-28T12:53:36Z" level=debug msg="job pod found [backrest-backup-cbcd70-pjjw2]"
time="2019-08-28T12:53:36Z" level=debug msg="[PodController] onUpdate ns=cfae-postgres-testing /api/v1/namespaces/cfae-postgres-testing/pods/backrest-backup-cbcd70-pjjw2"
time="2019-08-28T12:53:36Z" level=debug msg="[JobController] onUpdate ns=cfae-postgres-testing /apis/batch/v1/namespaces/cfae-postgres-testing/jobs/backrest-backup-cbcd70 active=1 succeeded=0 conditions=[[]]"
time="2019-08-28T12:53:36Z" level=debug msg="job controller backup label "
time="2019-08-28T12:53:36Z" level=debug msg="jobController onUpdate backrest job case"
time="2019-08-28T12:53:36Z" level=debug msg="got a backrest job status=0"
time="2019-08-28T12:53:36Z" level=debug msg="job pod found [backrest-backup-cbcd70-pjjw2]"
time="2019-08-28T12:53:38Z" level=debug msg="[PodController] onUpdate ns=cfae-postgres-testing /api/v1/namespaces/cfae-postgres-testing/pods/backrest-backup-cbcd70-pjjw2"
time="2019-08-28T12:53:39Z" level=debug msg="job pod found [backrest-backup-cbcd70-pjjw2]"
time="2019-08-28T12:54:21Z" level=debug msg="[PodController] onUpdate ns=cfae-postgres-testing /api/v1/namespaces/cfae-postgres-testing/pods/backrest-backup-cbcd70-pjjw2"
time="2019-08-28T12:54:21Z" level=debug msg="job pod found [backrest-backup-cbcd70-pjjw2]"
time="2019-08-28T12:54:21Z" level=debug msg="[JobController] onUpdate ns=cfae-postgres-testing /apis/batch/v1/namespaces/cfae-postgres-testing/jobs/backrest-backup-cbcd70 active=0 succeeded=1 conditions=[[{Complete True 2019-08-28 12:54:21 +0000 UTC 2019-08-28 12:54:21 +0000 UTC  }]]"
time="2019-08-28T12:54:21Z" level=debug msg="job controller backup label "
time="2019-08-28T12:54:21Z" level=debug msg="jobController onUpdate backrest job case"
time="2019-08-28T12:54:21Z" level=debug msg="got a backrest job status=1"
time="2019-08-28T12:54:21Z" level=debug msg="update the status to completed here for backrest cbcd70 job backrest-backup-cbcd70"
time="2019-08-28T12:54:21Z" level=debug msg="[{\"op\":\"replace\",\"path\":\"/spec/status\",\"value\":\"job completed\"}]"
time="2019-08-28T12:54:21Z" level=debug msg="[PgtaskController] onUpdate ns=cfae-postgres-testing /apis/crunchydata.com/v1/namespaces/cfae-postgres-testing/pgtasks/backrest-backup-cbcd70"
```

Second backup [Failed]:
```
$ pgo backup cbcd70 --backup-opts="--type=full" -n cfae-postgres-testing
created Pgtask backrest-backup-cbcd70

# job and pod
$ kubectl get jobs backrest-backup-cbcd70
NAME                     DESIRED   SUCCESSFUL   AGE
backrest-backup-cbcd70   1         1            23m

$ kubectl get pod backrest-backup-cbcd70-pjjw2
NAME                           READY     STATUS      RESTARTS   AGE
backrest-backup-cbcd70-pjjw2   0/1       Completed   0          25m

# logs
time="2019-08-28T13:15:06Z" level=error msg="error creating Job backrest-backup-cbcd70jobs.batch \"backrest-backup-cbcd70\" already exists"
time="2019-08-28T13:15:06Z" level=debug msg="[PgtaskController] onUpdate ns=cfae-postgres-testing /apis/crunchydata.com/v1/namespaces/cfae-postgres-testing/pgtasks/backrest-backup-cbcd70"
```

**Comment by ajoyej** [comment]
Hello,

Can anyone confirm this is a bug?
If so, any plan to fix this?

Thanks and regards,
José

**Comment by cahoonpwork** [comment]
I didn't reproduce the issue running manual backrest backups with 4.0.1.  How did you install the operator?  What version of the containers are you running?

**Comment by cahoonpwork** [comment]
I just reproduced this in 4.0.1 with the Ansible install trying to run multiple backups manually, not seeing it in the bash install.

**Comment by andrewlecuyer** [comment]
@ajoyej there is a defect in the Ansible installer for v4.0.1 that results in the behavior you are seeing.  Specifically, a label is missing from the `backrest-job.json` template that is used to create backup jobs that prevents the proper cleanup of a prior backup job when creating a new one (therefore preventing the ability to create a new backup).

This issue has been addressed in #907 and will be included in the next PGO release (v4.1.0).

For now, you can copy the template from the bash install location (where it has the proper label) to the proper location in the Ansible install, and then update or reinstall the PGO.  You can copy the template file to the proper location in the Ansible install as follows:

`cp $PGOROOT/conf/postgres-operator/backrest-job.json $PGOROOT/ansible/roles/pgo-operator/files/pgo-configs/backrest-job.json`

**Comment by ajoyej** [comment]
Thanks a lot for the workaround!

José
